packages/identifier-resolution/src/functions/externalOIDFIdentifier.ts

```diff
@@ -0,0 +1,93 @@
+ *
+ * @returns Promise resolving to an ExternalIdentifierOIDFEntityIdResult containing:
+ *  - trustedAnchors: Record mapping trust anchors to their public key hexes
+ *  - errorList: Optional record of errors encountered per trust anchor
+ *  - jwks: Array of JWK information from the trust chain
+ *  - trustEstablished: Boolean indicating if any trust relationships were established
+ *
+ * @throws Error if trust anchors are missing or JWT verification plugin is not enabled
+ */
+export async function resolveExternalOIDFEntityIdIdentifier(
+  opts: ExternalIdentifierOIDFEntityIdOpts,
+  context: IAgentContext<IOIDFClient>
+): Promise<ExternalIdentifierOIDFEntityIdResult> {
+  let { trustAnchors, identifier } = opts
+
+  if (!trustAnchors || trustAnchors.length === 0) {
+    return Promise.reject(Error('ExternalIdentifierOIDFEntityIdOpts is missing the trustAnchors'))
+  }
+
+  if (!contextHasPlugin(context, 'jwtVerifyJwsSignature')) {
+    return Promise.reject(Error('For OIDFEntityId resolving the agent needs to have the JwtService plugin enabled'))
+  }
+
+  const trustedAnchors: Record<TrustedAnchor, PublicKeyHex> = {}
+  const errorList: Record<TrustedAnchor, ErrorMessage> = {}
+  const jwkInfos: Array<ExternalJwkInfo> = []
+
+  for (const trustAnchor of trustAnchors) {
+    const resolveResult = await context.agent.resolveTrustChain({
+      entityIdentifier: identifier,
+      trustAnchors: [trustAnchor]
+    })
+
+    if (resolveResult.error || !resolveResult.trustChain) {
+      errorList[trustAnchor] = resolveResult.errorMessage ?? 'unspecified'
+    } else {
+      const trustChain: ReadonlyArray<string> = resolveResult.trustChain.asJsReadonlyArrayView()
+      if (trustChain.length === 0) {
+        errorList[trustAnchor] = 'Trust chain is empty'
+        continue
+      }
+      
+      const jwt = trustChain[trustChain.length - 1]
+      const jwtVerifyResult: IJwsValidationResult = await context.agent.jwtVerifyJwsSignature({ jws: jwt })
+
+      if (jwtVerifyResult.error || jwtVerifyResult.critical) {
+        errorList[trustAnchor] = jwtVerifyResult.message
+        continue
+      }
+
+      if (jwtVerifyResult.jws.signatures.length === 0) {
+        errorList[trustAnchor] = 'No signature was present in the trust anchor JWS'
+        continue
+      }
+
+      const signature = jwtVerifyResult.jws.signatures[0]
+      if (signature.identifier.jwks.length === 0) {
+        errorList[trustAnchor] = 'No JWK was present in the trust anchor signature'
+        continue
+      }
+
+      const jwkInfo: ExternalJwkInfo = signature.identifier.jwks[0]
+      jwkInfos.push(jwkInfo)
+      trustedAnchors[trustAnchor] = signature.publicKeyHex
+    }
+  }
+
+  return {
+    method: 'entity_id',
+    trustedAnchors,
+    ...(Object.keys(errorList).length > 0 && { errorList }),
+    jwks: jwkInfos,
+    trustEstablished: Object.keys(trustedAnchors).length > 0
```

Comment:
should we not look at errors here to determine if trust is established?

Reply:
No, most of the trust anchors will always produce errors. The entity will ony be added trustedAnchors when trust was established with one of the anchors.